lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -164,9 +166,64 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
         {DxcOutputObject::DataOutput(DXC_OUT_OBJECT, pResult, DxcOutNoName),
          DxcOutputObject::DataOutput(DXC_OUT_ERRORS, pErrorBlob, DxcOutNoName)},
         ppResult));
-    return S_OK;
   }
   CATCH_CPP_RETURN_HRESULT();
+
+  if (ppResult == nullptr || *ppResult == nullptr)
+    return S_OK;
+
+  HRESULT HR;
+  (*ppResult)->GetStatus(&HR);
+  if (FAILED(HR))
+    return HR;
+
+  CComPtr<IDxcBlob> pObject;
+  IFR((*ppResult)->GetResult(&pObject));
+
+  // Add Hash.
+  LPVOID PTR = pObject->GetBufferPointer();
+  if (IsDxilContainerLike(PTR, pObject->GetBufferSize()))
+    HashAndUpdate((DxilContainerHeader *)PTR);
+  return S_OK;
```

Comment:
This code should be located within the try/catch, before the DxcResult construction, in case any fatal errors/exceptions were to occur in this process, and to allow error messages to be captured if necessary.  It should use `pResult` directly instead of loading `pObject` back out of `ppResult`.  If `IsContainerLike` were to fail, it indicates a malformed container result, which is a fatal error that should cause this to abort, not just skip container hashing.